src/atomsbase_interface.jl

```diff
@@ -0,0 +1,80 @@
+#
+# Interface between AtomsBase and GeomOpt that provides 
+# utility functions for manipulating systems.
+#
+# This interface helps defining which particles in the system are considered 
+# optimizable. Note that by default all particles are assumed to be optimizable. 
+# The user can call clamp_atoms to fix atoms whose position should not be optimized.
+#
+# IMPORTANT: Note that we always work in cartesian coordinates.
+#
+export update_positions, update_optimizable_coordinates, clamp_atoms
+
+
+@doc raw"""
+    update_postions(system::AbstractSystem, positions::Vector{<:AbstractVector{<:Real}}) where {L <: Unitful.Length}
+
+Creates a new system based on ``system`` but with atoms positions updated to the ones specified in `positions`, 
+using cartesian coordinates.
+"""
+function update_positions(system::AbstractSystem, positions::AbstractVector{<:AbstractVector{L}}) where {L <: Unitful.Length}
+    particles = [
+                 Atom(atom, position=SVector{3,L}(position)) 
+                 for (atom, position) in zip(system.particles, positions)]
+    AbstractSystem(system, particles=particles)
+end
+
+@doc raw"""
+    update_optimizable_coordinates(system::AbstractSystem, positions::Vector{<:AbstractVector{<:L}}) where {L <: Unitful.Length}
+
+Creates a new system based on ``system`` with the optimizable coordinates are
+updated to the ones provided (in the order in which they appear in system.particles), cartesian coordinates version..
+"""
+function update_optimizable_coordinates(system::AbstractSystem, positions::AbstractVector{<:L}) where {L <: Unitful.Length}
+    mask = get_optimizable_mask(system)
+    new_positions = Vector.(position(system)) # make mutable.
+    new_positions[mask] = [eachcol(reshape(positions, 3, :))...]
```

Comment:
You could use `reinterpret` here to remove allocation

```julia
reinterpret(reshape, SVector{3, eltype(positions)}, reshape(positions, 3, :))
```

Also when allocating similar array you should use `similar`. So, `similar(position(system))` instead of `Vector.(position(system))`. Or, if you want to copy then `deepcopy(position(system))`.